src/math/p5.Vector.js

```diff
@@ -326,6 +326,129 @@ p5.Vector.prototype.add = function add(x, y, z) {
   return this;
 };
 
+/// HELPERS FOR REMAINDER METHOD
+const calculateRemainder2D = function(xComponent, yComponent) {
+  if (xComponent === 0 && yComponent === 0) {
+    return this;
+  } else if (xComponent === 0) {
+    this.y = this.y % x[1];
+    return this;
+  } else if (yComponent === 0) {
+    this.x = this.x % x[0];
+    return this;
+  } else {
+    this.x = this.x % xComponent;
+    this.y = this.y % yComponent;
+    return this;
+  }
+};
+
+const calculateRemainder3D = function(xComponent, yComponent, zComponent) {
+  if (xComponent === 0 && yComponent === 0 && zComponent === 0) {
+    return this;
+  } else if (xComponent === 0) {
```

Comment:
does this account correctly for the case where two of the three are 0?

Reply:
Yes I have covered the edge cases in commit 0b9302d